### DISCUSSION TITLE
MHW tracking with large datasets

### DISCUSSION THREAD

**Discussion by robertsmalcolm** [ocetrac/ocetrac#35] [General]
Hi. I really appreciate the package, it has the potential to be extremely useful. I'm trying to work out how I can use it with large datasets. For example, if I want to identify and track MHWs in daily global data, perhaps at 0.25 degree resolution or finer, with decades to centuries of data (from climate models). In that case, it is not possible to follow your example and read all the data at one time. I can perhaps figure out how to calculate the daily threshold for a climatology, but then the ideal would be to read 1 year of daily data, do MHW identification and tracking, store this, read the next year of data, identify and track MHW and add onto the previous year, hence building up a multi-year MHW database. I'm not sure if or how I could use this package in such as way (large datasets are a problem for other MHW methods too I think). Thanks.

**Comment by CassiaCai** [comment]
Hi Robert, hmm what is your current process of tracking and storing them? I think it should possible to do what you are describing. I have used Ocetrac with the CESM2 LE (which is although at a much lower resolution (1 degree)), there are 100 ensemble members and have done them separately and then resaved them on MHW ID. Perhaps I misunderstand. Sorry for only picking up the conversation now.
